File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> s;
        vector <int> B;
        for(int i=0;i<nums.size();i++)
        {
            if(s.find(target-nums[i])!=s.end())
            {
            
                B.push_back(i);
                B.push_back(s[target-nums[i]]);
                break;
            }
            s[nums[i]] = i;
            
        }
        return B;
        
    }
};

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int n = nums.length;
        HashMap<Integer,Integer> A = new HashMap<>();
        for(int i=0;i<n;i++)
        {
            if(A.containsKey(target - nums[i]))
                return new int[]{A.get(target - nums[i]),i};
            else
                A.put(nums[i],i);
            
        }
        return new int[]{-1,-1};
        
    }
}